src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java

```diff
@@ -846,11 +841,18 @@ public void deleteResultsAndSubmissionsOfParticipation(Participation participati
                     Submission submissionToDelete = result.getSubmission();
                     submissionRepository.deleteById(submissionToDelete.getId());
                     result.setSubmission(null);
-                    // make sure submissions don't get deleted twice (see below)
                     participation.removeSubmissions(submissionToDelete);
                 }
             }
         }
+        // The following case is necessary, because we might have submissions without a result. At this point only submissions without a result will still be connected to the
```

Comment:
@thilo-behnke Looks good to me! Just one minor remark: You have covered participations without a result here, so maybe it's good to have this case covered in the unit tests as well?

Reply:
You are right, I updated the test.